crates/rpc/src/handlers/prepare_beacon_proposer.rs

```diff
@@ -0,0 +1,116 @@
+
+    for item in items {
+        operation_pool.insert_proposer_preparation(item.validator_index, item.fee_recipient);
+    }
+
+    Ok(HttpResponse::Ok().finish())
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use actix_web::{App, http::StatusCode, test};
+    use alloy_primitives::Address;
+
+    use super::*;
+
+    #[actix_web::test]
+    async fn test_prepare_beacon_proposer_success() {
+        let operation_pool = Arc::new(OperationPool::default());
+        let app = test::init_service(
+            App::new()
+                .app_data(Data::new(operation_pool))
+                .service(prepare_beacon_proposer),
+        )
+        .await;
+
+        let fee_recipient = Address::from([0x42; 20]);
+        let items = vec![
+            PrepareBeaconProposerItem {
+                validator_index: 1,
+                fee_recipient,
+            },
+            PrepareBeaconProposerItem {
+                validator_index: 2,
+                fee_recipient,
+            },
+        ];
+
+        let req = test::TestRequest::post()
+            .uri("/eth/v1/validator/prepare_beacon_proposer")
+            .set_json(&items)
+            .to_request();
+
+        let resp = test::call_service(&app, req).await;
+        assert_eq!(resp.status(), StatusCode::OK);
+    }
+
+    #[actix_web::test]
+    async fn test_prepare_beacon_proposer_empty_request() {
+        let operation_pool = Arc::new(OperationPool::default());
+        let app = test::init_service(
+            App::new()
+                .app_data(Data::new(operation_pool))
+                .service(prepare_beacon_proposer),
+        )
+        .await;
+
+        let items: Vec<PrepareBeaconProposerItem> = vec![];
+
+        let req = test::TestRequest::post()
+            .uri("/eth/v1/validator/prepare_beacon_proposer")
+            .set_json(&items)
+            .to_request();
+
+        let resp = test::call_service(&app, req).await;
+        assert_eq!(resp.status(), StatusCode::BAD_REQUEST);
+    }
+
+    #[actix_web::test]
+    async fn test_prepare_beacon_proposer_stores_data() {
+        let operation_pool = Arc::new(OperationPool::default());
+        let app = test::init_service(
+            App::new()
+                .app_data(Data::new(operation_pool.clone()))
+                .service(prepare_beacon_proposer),
+        )
+        .await;
+
+        let fee_recipient = Address::from([0x42; 20]);
+        let items = vec![PrepareBeaconProposerItem {
+            validator_index: 123,
+            fee_recipient,
+        }];
+
+        let req = test::TestRequest::post()
+            .uri("/eth/v1/validator/prepare_beacon_proposer")
+            .set_json(&items)
+            .to_request();
+
+        let resp = test::call_service(&app, req).await;
+        assert_eq!(resp.status(), StatusCode::OK);
+
+        let stored_fee_recipient = operation_pool.get_proposer_preparation(123);
+        assert_eq!(stored_fee_recipient, Some(fee_recipient));
```

Comment:
```suggestion
        assert_eq!(operation_pool.get_proposer_preparation(123), Some(fee_recipient));
```

can we inline this, as the variable is only used once